### ISSUE TITLE
Expose contention data through the Profiling APIs 

### ISSUE THREAD

**Issue by lt72** [dotnet/coreclr#7705]
Currently not exposed.

**Comment by noahfalk** [comment]
contention data for what thing? : ) 
Understanding the scenario motivating this request would be very useful.

**Comment by lt72** [comment]
This refers to exposing contention data as ETW does:
https://msdn.microsoft.com/en-us/library/ff356157%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396 

@mjsabby: please confirm

**Comment by mjsabby** [comment]
yup that's the one. This is to be able to get this data without using ETW.

**Comment by lt72** [comment]
Please let us know if this is still a concern and what the time frame is that you care about.

**Comment by gleocadie** [comment]
Hi, this is the kind of thing we would like to have. Lately, we add a contention issue on Linux. I added the contention support in the Profiling API and with a custom build, we were able to get stack traces and find our contention issue. There might be other (better?) solution, but at that time, I was not aware (still don't) and this was the easiest way for us.

I have the code ready, will you be interested in a PR?
